src/utils/tabs.rs

```diff
@@ -0,0 +1,74 @@
+use crate::config::SPACES;
+
+use unicode_segmentation::UnicodeSegmentation;
+
+pub fn clamp_arg(tab_size: usize) -> usize {
+    std::cmp::min(tab_size, SPACES.len())
+}
+
+/// Expand tabs as spaces.
+/// tab_width = 0 is documented to mean do not replace tabs.
+pub fn expand(line: &str, tab_width: usize) -> String {
+    let line_bytes = line.as_bytes();
+    if tab_width == 0 || !line_bytes.iter().any(|c| *c == b'\t') {
+        line.to_string()
+    } else {
+        let expanded = Vec::with_capacity(line.len());
+
+        let expand_tabs = |mut v: Vec<u8>, c: &u8| -> Vec<u8> {
+            if *c == b'\t' {
+                v.extend_from_slice(SPACES[..tab_width].as_bytes());
```

Comment:
I may be missing something obvious but why not a simple split + join ?
```rust
line.split('\t').collect::<Vec<_>>().join(&SPACES[..tab_width])
```

Or do we want to improve expansion and make it expand as tab expand in editor (aka not a fixed length but they extend up to a modulo

Something like this:
```rust
let (expanded, _len): (String, _) = line.chars().fold(("".into(), 0), | (s, len), e| {
        if e != '\t' {
            (s + &e.to_string(), len + 1)
        } else {
            let tab_len = tab_width - (len % tab_width)
        ;
            (s + &SPACES[..tab_len], len + tab_len)
        }
    });
```